### PR TITLE
feat!: update middleware contracts version

### PR DESCRIPTION
**Pull request by maximopalopoli**
### What Changed?
This PR updates middleware contracts version to [v1.1.1-testnet-slashing release](https://github.com/Layr-Labs/eigenlayer-middleware/releases/tag/v1.1.1-testnet-slashing).

Main changes:
- A tests is commented due to a change in the churn registration interface (now the churn option is set in the params, but uses the registerForOperatorSet method).
- Added AVSRegistrar get and set methods in elcontracts reader and writer
- Removed IsOperatorSetQuorum function from avsRegistry reader
- Changed behavior in IsOperatorRegisteredWithOperatorSet method (elcontracts reader): Previously, in case the operator set id, the query was directed to AVSDirectory, but now is always directed to allocationManager.

### Reviewer Checklist
- [ ] Code is well-documented
- [ ] Code adheres to Go [naming conventions](https://go.dev/doc/effective_go#names)
- [ ] Code deprecates any old functionality before removing it

### CODE REVIEW

File: chainio/clients/elcontracts/reader.go [hunk 1]
```diff
@@ -819,36 +831,21 @@ func (r *ChainReader) IsOperatorRegisteredWithOperatorSet(
...
-	if operatorSet.Id == 0 {
-		// this is an M2 AVS
-		if r.avsDirectory == nil {
-			return false, errors.New("AVSDirectory contract not provided")
-		}
-
-		status, err := r.avsDirectory.AvsOperatorStatus(&bind.CallOpts{Context: ctx}, operatorSet.Avs, operatorAddress)
-		// This call should not fail since it's a getter
-		if err != nil {
-			return false, err
```
Comment (MegaRedHand) [inline chainio/clients/elcontracts/reader.go]: Can we add a new `IsOperatorRegisteredWithAvs` function with this functionality?

File: services/bls_aggregation/blsagg_test.go [hunk 2]
```diff
@@ -1620,15 +1622,40 @@ func TestIntegrationBlsAgg(t *testing.T) {
...
+			OperatorAddress: operatorAddr,
+			AVSAddress:      contractAddrs.ServiceManager,
+			OperatorSetIds:  quorumNumbersU32,
+			WaitForReceipt:  true,
+			Socket:          "socket",
+			BlsKeyPair:      otherKeyPair,
+		}
+
+		// Register operator
+		receipt, err := elcontractsWriter.SetAVSRegistrar(
```
Comment (MegaRedHand) [inline services/bls_aggregation/blsagg_test.go]: Let's move this to the Solidity deployment script.

File: CHANGELOG.md [hunk 3]
```diff
@@ -119,8 +121,11 @@ Each version will have a separate `Breaking Changes` section as well. To describ
     ).WithWindowDuration(windowDuration)
     blsAggServ.InitializeNewTask(metadata)
     ```
+* fix: `IsOperatorRegisteredWithOperatorSet` only queries to AVSDirectory by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
```
Comment (MegaRedHand) [inline CHANGELOG.md]: Let's mention `IsOperatorRegisteredWithAvs` as an alternative for M2 quorum users.

File: chainio/clients/elcontracts/reader.go (hunk 1 above)
Comment (maximopalopoli) [reply chainio/clients/elcontracts/reader.go]: Done in [8e75882](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/8e7588246abb42e3800ca4c22fa548d19d8b5496)

File: CHANGELOG.md (hunk 3 above)
Comment (maximopalopoli) [reply CHANGELOG.md]: Done in [63ef39f](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/63ef39fd5a2a750eea5ee3794b14858d69cc25c8)

File: services/bls_aggregation/blsagg_test.go (hunk 2 above)
Comment (maximopalopoli) [reply services/bls_aggregation/blsagg_test.go]: Done in [Set avs registrar in deploy instead of doing it on every registration](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/a343e347cabfb625369885d57eccb8f9b1b62395)

File: CHANGELOG.md [hunk 4]
```diff
@@ -37,7 +37,8 @@ Each version will have a separate `Breaking Changes` section as well. To describ
 
 * Added field `DontUseAllocationManager` to `BuildAllConfig` by @MegaRedHand in [#580](https://github.com/Layr-Labs/eigensdk-go/pull/580)
 * Added `AnvilC` field to `clients.Clients` struct by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
-* Added `GetAVSRegistrar` and `SetAVSRegistrar` methods to elcontracts chain reader and chain writer by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
+* Added `IsOperatorRegisteredWithAvs`, `GetAVSRegistrar` methods to elcontracts chain reader and and `SetAVSRegistrar` to chain writer by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
```
Comment (MegaRedHand) [inline CHANGELOG.md:40]: Let's add some examples here

File: CHANGELOG.md [hunk 5]
```diff
@@ -121,7 +122,8 @@ Each version will have a separate `Breaking Changes` section as well. To describ
     ).WithWindowDuration(windowDuration)
     blsAggServ.InitializeNewTask(metadata)
     ```
-* fix: `IsOperatorRegisteredWithOperatorSet` only queries to AVSDirectory by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
+* The elcontracts `chainReader.IsOperatorRegisteredWithOperatorSet` does not query to AVSDirectory anymore, only to allocationManager by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
+  * The query to AVSDirectory i still available using `chainReader.IsOperatorRegisteredWithAvs` method
```
Comment (MegaRedHand) [inline CHANGELOG.md]:
```suggestion
  * The query to AVSDirectory is still available using `chainReader.IsOperatorRegisteredWithAvs` method
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -121,7 +122,8 @@ Each version will have a separate `Breaking Changes` section as well. To describ
     ).WithWindowDuration(windowDuration)
     blsAggServ.InitializeNewTask(metadata)
     ```
-* fix: `IsOperatorRegisteredWithOperatorSet` only queries to AVSDirectory by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
+* The elcontracts `chainReader.IsOperatorRegisteredWithOperatorSet` does not query to AVSDirectory anymore, only to allocationManager by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
```
Comment (MegaRedHand) [inline CHANGELOG.md]:
```suggestion
* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`. It now only queries the `AllocationManager` by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
```

File: chainio/clients/avsregistry/writer_test.go [hunk 7]
```diff
@@ -560,25 +533,24 @@ This test is commented because produces an edge case that leads to a bug
...
 	}
 
+	receipt, err := elWriter.RegisterForOperatorSets(context.Background(), contractAddrs.RegistryCoordinator, request)
+	require.NoError(t, err)
+	require.NotNil(t, receipt)
 
-		receipt, err = elWriter.RegisterForOperatorSets(context.Background(), contractAddrs.RegistryCoordinator, request)
-		require.NoError(t, err)
-		require.NotNil(t, receipt)
+	isRegisterd, err = chainReader.IsOperatorRegistered(&bind.CallOpts{}, operatorAddr)
```
Comment (MegaRedHand) [inline chainio/clients/avsregistry/writer_test.go]: Typo:
```suggestion
	isRegistered, err = chainReader.IsOperatorRegistered(&bind.CallOpts{}, operatorAddr)
```

File: chainio/clients/elcontracts/reader.go [hunk 8]
```diff
@@ -823,9 +823,9 @@ func (r *ChainReader) GetOperatorSetsForOperator(
 	return r.allocationManager.GetAllocatedSets(&bind.CallOpts{Context: ctx}, operatorAddress)
 }
 
-// Returns `true` if an operator is registered with a specific operator set or M2 quorum.
-// Can return an error if the `AVSDirectory` or `AllocationManager` contract addresses were
-// not provided, or due to errors in the underlying contract call.
+// Returns `true` if an operator is registered with a specific operator set. Can return an
+// error if the `AllocationManager` contract addresses was not provided, or due to errors
+// in the underlying contract call.
```
Comment (MegaRedHand) [inline chainio/clients/elcontracts/reader.go:828]: Let's mention this does not work with M2 quorums.

File: chainio/clients/elcontracts/reader.go [hunk 9]
```diff
@@ -848,6 +848,27 @@ func (r *ChainReader) IsOperatorRegisteredWithOperatorSet(
 	return false, nil
 }
 
+// Returns `true` if an operator is registered with a specific M2 quorum, querying AVSDirectory.
+// Can return an error if the `AVSDirectory` contract addres was not provided, or due to errors
+// in the underlying contract call.
+func (r *ChainReader) IsOperatorRegisteredWithAvs(
+	ctx context.Context,
+	operatorAddress gethcommon.Address,
+	operatorSet allocationmanager.OperatorSet,
```
Comment (MegaRedHand) [inline chainio/clients/elcontracts/reader.go]: Let's change this to receive the AVS address instead of this.

File: contracts/script/DeployMockAvsRegistries.s.sol [hunk 10]
```diff
@@ -129,19 +129,17 @@ contract DeployMockAvsRegistries is Script, ConfigsReadWriter, EigenlayerContrac
...
-                IServiceManager(address(manager)),
-                registries.stakeRegistry,
-                registries.blsApkRegistry,
-                registries.indexRegistry,
-                registries.socketRegistry,
-                eigen.allocationManager,
-                deployed.pauserReg
-            )
+        deployed.coordinatorImplementation = new SlashingRegistryCoordinator(
+            //IServiceManager(address(manager)),
```
Comment (MegaRedHand) [inline contracts/script/DeployMockAvsRegistries.s.sol]: Let's remove this comment

File: contracts/script/DeployMockAvsRegistries.s.sol [hunk 11]
```diff
@@ -160,8 +158,8 @@ contract DeployMockAvsRegistries is Script, ConfigsReadWriter, EigenlayerContrac
                 (config.communityMultisig, config.churner, config.ejector, 0, address(manager))
             )
         );
-        LegacyRegistryCoordinator(address(deployed.coordinator)).enableM2QuorumRegistration();
-        LegacyRegistryCoordinator(address(deployed.coordinator)).disableOperatorSets();
+        // LegacyRegistryCoordinator(address(deployed.coordinator)).enableM2QuorumRegistration();
+        // LegacyRegistryCoordinator(address(deployed.coordinator)).disableOperatorSets();
```
Comment (MegaRedHand) [inline contracts/script/DeployMockAvsRegistries.s.sol]: Same

File: services/bls_aggregation/blsagg_test.go [hunk 12]
```diff
@@ -1009,8 +1009,10 @@ func TestBlsAgg(t *testing.T) {
 			taskResponseDigest2, err := hashFunction(taskResponse2)
 			require.Nil(t, err)
 			blsSigOp2 := testOperator2BlsKeypair.SignMessage(taskResponseDigest2)
-			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Nanosecond)
 			defer cancel()
+			time.Sleep(1 * time.Millisecond)
```
Comment (MegaRedHand) [inline services/bls_aggregation/blsagg_test.go]: Let's just use a duration of 0.
```suggestion
			ctx, cancel := context.WithTimeout(context.Background(), 0)
			defer cancel()
```

File: CHANGELOG.md (hunk 4 above)
Comment (maximopalopoli) [reply CHANGELOG.md:40]: Done in [82d0647](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/82d0647aff916052767b8e0d4d50a5b43aad84ea)

File: CHANGELOG.md (hunk 5 above)
Comment (maximopalopoli) [reply CHANGELOG.md]: Done in [1412b2c](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/1412b2cd0c6afe3df2e3340503cabe669b52260f)

File: CHANGELOG.md (hunk 6 above)
Comment (maximopalopoli) [reply CHANGELOG.md]: Done in [1412b2c](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/1412b2cd0c6afe3df2e3340503cabe669b52260f)

File: chainio/clients/avsregistry/writer_test.go (hunk 7 above)
Comment (maximopalopoli) [reply chainio/clients/avsregistry/writer_test.go]: Done in [ac31d72](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/ac31d72a812cba1821e4a027753fdce6a7cae209)

File: chainio/clients/elcontracts/reader.go (hunk 8 above)
Comment (maximopalopoli) [reply chainio/clients/elcontracts/reader.go:828]: Done in [b9cdc7a](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/b9cdc7af1c6a70ee4c37954af46536b465084cfb)

File: chainio/clients/elcontracts/reader.go (hunk 9 above)
Comment (maximopalopoli) [reply chainio/clients/elcontracts/reader.go]: Done in [7291057](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/729105771edee8dda3f322186bb644dadb722119)

File: contracts/script/DeployMockAvsRegistries.s.sol (hunk 10 above)
Comment (maximopalopoli) [reply contracts/script/DeployMockAvsRegistries.s.sol]: Done in [4d52704](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/4d5270489f2d38a5d703a088a09d9c418a866020)

File: contracts/script/DeployMockAvsRegistries.s.sol (hunk 11 above)
Comment (maximopalopoli) [reply contracts/script/DeployMockAvsRegistries.s.sol]: Also done in [4d52704](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/4d5270489f2d38a5d703a088a09d9c418a866020)

File: services/bls_aggregation/blsagg_test.go (hunk 12 above)
Comment (maximopalopoli) [reply services/bls_aggregation/blsagg_test.go]: Done in [797861f](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/797861fb63fd2c0666a2bbf68ddf93e72274601e)

File: CHANGELOG.md [hunk 13]
```diff
@@ -122,8 +141,8 @@ Each version will have a separate `Breaking Changes` section as well. To describ
     ).WithWindowDuration(windowDuration)
     blsAggServ.InitializeNewTask(metadata)
     ```
-* The elcontracts `chainReader.IsOperatorRegisteredWithOperatorSet` does not query to AVSDirectory anymore, only to allocationManager by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
-  * The query to AVSDirectory i still available using `chainReader.IsOperatorRegisteredWithAvs` method
+* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`. It now only queries the `AllocationManager` by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
+  * The query to AVSDirectory is still available using `chainReader.IsOperatorRegisteredWithAvs` method
```
Comment (MegaRedHand) [inline CHANGELOG.md]: Let's rewrite this to mention `IsOperatorRegisteredWithOperatorSet` only works for operator sets, while `IsOperatorRegisteredWithAvs` works for m2 quorums

File: chainio/clients/elcontracts/reader.go [hunk 14]
```diff
@@ -851,16 +852,17 @@ func (r *ChainReader) IsOperatorRegisteredWithOperatorSet(
 // Returns `true` if an operator is registered with a specific M2 quorum, querying AVSDirectory.
 // Can return an error if the `AVSDirectory` contract addres was not provided, or due to errors
 // in the underlying contract call.
+// Note: this method does not work for not M2 quorums
```
Comment (MegaRedHand) [inline chainio/clients/elcontracts/reader.go]:
```suggestion
// Note: this method does not take into account operator sets
```

File: chainio/clients/elcontracts/reader.go [hunk 15]
```diff
@@ -826,6 +826,7 @@ func (r *ChainReader) GetOperatorSetsForOperator(
 // Returns `true` if an operator is registered with a specific operator set. Can return an
 // error if the `AllocationManager` contract addresses was not provided, or due to errors
 // in the underlying contract call.
+// Note: this method does not work for M2 quorums
```
Comment (MegaRedHand) [inline chainio/clients/elcontracts/reader.go]:
```suggestion
// Note: this method does not take into account M2 quorums
```

File: cmd/egnaddrs/main.go [hunk 16]
```diff
@@ -111,18 +113,25 @@ func getRegCoordAndServiceMngrAddr(
...
+				client,
+			)
+			if err != nil {
+				return common.Address{}, common.Address{}, err
+			}
+
+			serviceManagerAddr, err := registryCoordinatorC.ServiceManager(&bind.CallOpts{})
+			if err != nil {
+				return registryCoordinatorAddr, common.Address{}, nil
+			}
```
Comment (MegaRedHand) [inline cmd/egnaddrs/main.go]: Please add a comment explaining why we ignore this error.

File: CHANGELOG.md
Comment (MegaRedHand) [inline CHANGELOG.md:1]: Let's mention `egnaddrs` was updated to work with slashing AVSs and now returns `0x00..00` when it fails to fetch the ServiceManager from the RegistryCoordinator. Also, can we add some of this information to the command's "help" documentation?

File: CHANGELOG.md (hunk 13 above)
Comment (maximopalopoli) [reply CHANGELOG.md]: Done in [3e2b71e](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/3e2b71e9eac5e83856dbbd9aba8c27a1975024b0)

File: chainio/clients/elcontracts/reader.go (hunk 14 above)
Comment (maximopalopoli) [reply chainio/clients/elcontracts/reader.go]: Done in [cd09f5a](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/cd09f5aad2ba0b0e4827162829abb3354297267b)

File: chainio/clients/elcontracts/reader.go (hunk 15 above)
Comment (maximopalopoli) [reply chainio/clients/elcontracts/reader.go]: Also done in [cd09f5a](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/cd09f5aad2ba0b0e4827162829abb3354297267b)

File: cmd/egnaddrs/main.go (hunk 16 above)
Comment (maximopalopoli) [reply cmd/egnaddrs/main.go]: Done in [b1298c8](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/b1298c82d894a6e68019bb5ca0b8f246abf7cc59)

File: CHANGELOG.md
Comment (maximopalopoli) [reply CHANGELOG.md:1]: Done in [6649ce8](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/6649ce86ca9cd78907521a2e8165b0c5444a13b4) and [6f2b69c](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/6f2b69ca8b0f9c15aa1004bd43c753e0e6c30db3)

File: CHANGELOG.md [hunk 17]
```diff
@@ -141,8 +141,9 @@ Each version will have a separate `Breaking Changes` section as well. To describ
     ).WithWindowDuration(windowDuration)
     blsAggServ.InitializeNewTask(metadata)
     ```
-* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`. It now only queries the `AllocationManager` by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
-  * The query to AVSDirectory is still available using `chainReader.IsOperatorRegisteredWithAvs` method
+* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`, only `AllocationManager`, since only works for operator sets by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
+  * To query if an operator is register to an m2 quorum you should now use `chainReader.IsOperatorRegisteredWithAvs` method, that queries the AVSDirectory.
+* `egnaddrs` utility now works with slashing version, and in case the registryCoordinator is the slashing version and service manager address is not passed by flag, then the returned service manager address will be address zero (0x00..00)
```
Comment (MegaRedHand) [inline CHANGELOG.md]:
```suggestion
* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`, and so now only works for operator sets by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
  * To query if an operator is registered to an M2 quorum you should now use `chainReader.IsOperatorRegisteredWithAvs`, which queries the `AVSDirectory`.
* `egnaddrs` utility now works with slashing release middleware contracts and, in that case, the returned service manager will be the zero address, unless the `--service-manager` is specified.
```

File: cmd/egnaddrs/main.go [hunk 18]
```diff
@@ -126,6 +126,9 @@ func getRegCoordAndServiceMngrAddr(
 
 			serviceManagerAddr, err := registryCoordinatorC.ServiceManager(&bind.CallOpts{})
 			if err != nil {
+				// Note: We ignore this error because the slashing version of RegistryCoordinator does
+				// not have ServiceManager() method, but previos versions does, so not necessarily an
+				// error here should be fatal.
```
Comment (MegaRedHand) [inline cmd/egnaddrs/main.go]:
```suggestion
				// We assume a failure here means it's a `SlashingRegistryCoordinator`, which doesn't have a `serviceManager()` method.
				// In that case, we just return a zero address for the `ServiceManager`.
```

**Review by MegaRedHand — DISMISSED** [review]
LGTM

**Review by MegaRedHand — APPROVED** [review]
